jms/org.wso2.carbon.transport.jms/src/main/java/org/wso2/carbon/transport/jms/factory/CachedJMSConnectionFactory.java

```
@@ -0,0 +1,277 @@
+    }
+
+    @Override public void closeMessageProducer(MessageProducer messageProducer) {
+        closeProducer(messageProducer, false);
+    }
+
+    @Override public void closeSession(Session session) {
+        closeSession(session, false);
+    }
+
+    private boolean closeConnection(Connection connection, boolean forcefully) {
+        try {
+            if (this.cacheLevel < JMSConstants.CACHE_CONNECTION || forcefully) {
+                connection.close();
+            }
+        } catch (JMSException e) {
+            logger.error("JMS Exception while closing the connection.");
+        }
+        return false;
+    }
+
+    private boolean closeConsumer(MessageConsumer messageConsumer, boolean forcefully) {
+        try {
+            if (this.cacheLevel < JMSConstants.CACHE_CONSUMER || forcefully) {
+                messageConsumer.close();
+            }
+        } catch (JMSException e) {
+            logger.error("JMS Exception while closing the consumer.");
+        }
+        return false;
+    }
+
+    private boolean closeProducer(MessageProducer messageProducer, boolean forcefully) {
+        try {
+            if (this.cacheLevel < JMSConstants.CACHE_PRODUCER || forcefully) {
+                messageProducer.close();
+            }
+        } catch (JMSException e) {
+            logger.error("JMS Exception while closing the producer.");
+        }
+        return false;
+    }
+
+    private boolean closeSession(Session session, boolean forcefully) {
+        try {
+            if (this.cacheLevel < JMSConstants.CACHE_SESSION || forcefully) {
+                session.close();
+            }
+        } catch (JMSException e) {
+            logger.error("JMS Exception while closing the consumer.");
+        }
+        return false;
+    }
+
+    private void resetCache() {
+        if (cachedMessageConsumer != null) {
+            try {
+                cachedMessageConsumer.close();
+            } catch (JMSException e) {
```

The exception is swallowed. Please correct in all places

---

Have fixed it now